internal/apis/certmanager/v1alpha2/types_certificate.go

```diff
@@ -275,6 +275,15 @@ type X509Subject struct {
 	// Serial number to be used on the Certificate.
 	// +optional
 	SerialNumber string `json:"serialNumber,omitempty"`
+	// Extra names to be used on the Certificate.
+	ExtraNames []ExtraName `json:"extraNames,omitempty"`
```

Comment:
Should this be under the 'subject' field? Also curious as to how this will relate to the other 'RawSubject' proposal that's being looked into too 👀

Reply:
I'm not aware of the "RawSubject" proposal. Is there a link I can review about that? I just want to know why that's needed as I don't see a need for it if we're allowing users to use OIDs directly. Also, that sounds like the validation would be even trickier than using this method.

Reply:
https://github.com/cert-manager/cert-manager/pull/5002 there you go 😄 I think a key request there is around how the subject is actually rendered, and the ordering of the components of it

Reply:
Ok, that makes sense. I think there will be some merge conflicts, but if the rawsubject is provided, then it should skip this ExtraNames field. It's interesting that the ordering matters and not just the ordering, but also how it's grouped. It sounds like that would still be useful for those wanting complete control. I also think there should be some description for what that field is in the PR and how it works. I'll leave a comment about it.

Reply:
On second thought... if it is empty, i think it should show as empty, so yes, deleting both omitEmpty annotations is probably correct.

Reply:
>Should this be under the 'subject' field?

I think so as this part of the subject, just like `ExtraNames` in are under the `Name` in the pkix package.

Reply:
Agreed. At first, I tried putting it where CommonName was, at the top, but while coding it this way, it didn't feel right. 

To give context, I tried this both ways, both under Subject and under CertificateSpec and now I feel If anything, CommonName being a top-level field is weird. I get it that it's kind of a first-class citizen in the world of certificates, but it really should be part of the subject as it's not truly a top-level field when you look at a certificate.

I had to have some code to map between the "Subject" of cert-manager into the "Subject" of pkix.Name and back because the pkix library code's structure follows the layout of a human-readable certificate where all "Subject" fields are inside pkix.Name.